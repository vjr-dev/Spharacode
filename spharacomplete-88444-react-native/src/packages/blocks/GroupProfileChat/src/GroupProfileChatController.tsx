// Customizable Area Start
//@ts-nocheck
//@ts-ignore
import { IBlock } from "framework/src/IBlock";
import { Message } from "framework/src/Message";
import { BlockComponent } from "framework/src/BlockComponent";
import MessageEnum, { getName } from "framework/src/Messages/MessageEnum";
import { runEngine } from "framework/src/RunEngine";

export const configJSON = require("./config");

export interface Props {
  navigation: any;
  id: string;
}

interface S {
  modal2Visible: boolean;
  switch1: boolean;
  switch2: boolean;
  Arr_Emergancy: any;
}

interface SS {
  id: any;
}

export default class GroupProfileChatController extends BlockComponent<
  Props,
  S,
  SS
> {
  constructor(props: Props) {
    super(props);
    this.receive = this.receive.bind(this);
    this.subScribedMessages = [getName(MessageEnum.AccoutLoginSuccess)];

    this.state = {
      modal2Visible: false,
      switch1: false,
      switch2: false,
      Arr_Emergancy: [
        {
          id: 1,
          img: "https://randomuser.me/api/portraits/men/84.jpg",
          name: "James",
        },
        {
          id: 2,
          img: "https://randomuser.me/api/portraits/men/43.jpg",
          name: "Prince",
        },
        {
          id: 3,
          img: "https://randomuser.me/api/portraits/men/35.jpg",
          name: "Henry",
        },
        {
          id: 4,
          img: "https://randomuser.me/api/portraits/men/19.jpg",
          name: "Charls",
        },
        {
          id: 5,
          img: "https://randomuser.me/api/portraits/men/92.jpg",
          name: "David",
        },
      ],
    };

    runEngine.attachBuildingBlock(this as IBlock, this.subScribedMessages);
  }

  async receive(from: string, message: Message) {
    runEngine.debugLog("Message Recived", message);
  }

  PluseClick1() {
    console.log("PLUS");
  }
}
// Customizable Area End
