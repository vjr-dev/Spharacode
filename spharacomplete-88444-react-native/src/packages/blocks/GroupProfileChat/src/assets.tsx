// Customizable Area Start
export const back1 = require("../assets/image_background-1.png");
export const back2 = require("../assets/image_background-2.png");
export const image_back = require("../assets/image_back.png");
export const Menu = require("../assets/Menu.png");
export const Pin = require("../assets/Pin.png");
export const Delete = require("../assets/Delete.png");
export const Archive = require("../assets/Archive.png");
export const More = require("../assets/More.png");
export const User = require("../assets/profile.png");
export const video = require("../assets/video.png");
export const call = require("../assets/call.png");
export const share = require("../assets/share.png");
export const option = require("../assets/option.png");
export const Next = require("../assets/next.png");
// Customizable Area End