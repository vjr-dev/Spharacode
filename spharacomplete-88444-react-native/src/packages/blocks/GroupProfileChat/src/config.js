// Customizable Area Start
Object.defineProperty(exports, "__esModule", {
    value: true
  });
  
  exports.validationApiContentType = "application/json";
  exports.validationApiMethodType = "GET";
  exports.exampleAPiEndPoint = "EXAMPLE_URL";
  exports.exampleAPiMethod = "POST";
  exports.exampleApiContentType = "application/json";
  exports.textInputPlaceHolder = "Enter Text";
  exports.labelTitleText = "PrivateChat";
  exports.labelBodyText = "PrivateChat Body";
  
  exports.btnExampleTitle = "CLICK ME";
  // Customizable Area End