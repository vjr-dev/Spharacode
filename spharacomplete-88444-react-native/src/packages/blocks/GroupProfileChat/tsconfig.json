{
    "extends": "../../../tsconfig.base.json",
    "compilerOptions": {
      "composite": true,
      "declaration": true,
      "emitDeclarationOnly": true,
      "jsx": "preserve",
      "isolatedModules": false,
      "outDir": "../../../dist",
      "rootDir": "../../../packages",
      "types": [ "node" ],
      "typeRoots": [
        "@types",
        "../../node_modules/@types"
      ],
      "strict": true,
      "paths": {
        "react": [ "./node_modules/@types/react" ]
      }
    },
    "include": ["../../blocks/**/src/**/*", "../../framework/src/**/*", "../../components/src/**/*"],
    "exclude": [
        "__tests__",
        "node_modules",
        "dist",
        "build",
        "scripts",
        "acceptance-tests",
        "webpack",
        "jest",
        "src/setupTests.ts"
      ],

  }
  
