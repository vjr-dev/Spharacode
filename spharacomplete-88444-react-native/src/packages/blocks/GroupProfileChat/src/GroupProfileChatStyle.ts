// Customizable Area Start
import { Dimensions, StyleSheet } from "react-native";
import { RFValue } from "react-native-responsive-fontsize";
import { COLORS } from "framework/src/Globals";
const sh = Dimensions.get("window").height;
const sw = Dimensions.get("window").width;

export const Styles = StyleSheet.create({
  container: {
    height: sh,
    width: sw,
    flex: 1,
  },
  safeareaView: {
    flex: 1,
    backgroundColor: "#454545",
  },
  modalView: {
    position: "absolute",
    bottom: 15,
    width: sw - 50,
    alignSelf: "center",
  },
  menuView: {
    width: "100%",
    borderRadius: 15,
    backgroundColor: "#4a4a4a",
    justifyContent: "center",
    alignItems: "center",
  },
  menuText: {
    padding: 15,
    fontSize: RFValue(15),
    color: "#bebebe",
  },
  seperator: {
    height: 1,
    backgroundColor: COLORS.skipGray,
    width: "90%",
  },
  cancelButton: {
    justifyContent: "center",
    alignItems: "center",
    width: sw - 50,
    backgroundColor: "#fed99f",
    borderRadius: 25,
  },
  cancelText: {
    fontSize: RFValue(13),
  },
  shadowView: {
    backgroundColor: "#3e3e3e",
    width: sw,
    height: sh,
    position: "absolute",
    opacity: 0.6,
    flex: 1,
  },
});
// Customizable Area End
