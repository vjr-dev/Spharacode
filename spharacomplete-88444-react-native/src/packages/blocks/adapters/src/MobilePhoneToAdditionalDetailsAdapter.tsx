import { v4 as uuidv4 } from 'uuid';
import { runEngine } from '../../../framework/src/RunEngine';
import { IBlock } from '../../../framework/src/IBlock';
import { Message } from '../../../framework/src/Message';
import MessageEnum, {
  getName
} from '../../../framework/src/Messages/MessageEnum';

export default class MobilePhoneToAdditionalDetailsAdapter {
  send: (message: Message) => void;

  constructor() {
    const blockId = uuidv4();
    this.send = message => runEngine.sendMessage(blockId, message);
    runEngine.attachBuildingBlock(this as IBlock, [
      getName(MessageEnum.NavigationMobilePhoneAdditionalDetailsMessage)
    ]);
  }

  convert = (from: Message): Message => {
    const to = new Message(getName(MessageEnum.NavigationMessage));

    to.addData(
      getName(MessageEnum.NavigationTargetMessage),
      'AdditionalDetailForm'
    );

    to.addData(
      getName(MessageEnum.NavigationPropsMessage),
      from.getData(getName(MessageEnum.NavigationPropsMessage))
    );

    const raiseMessage: Message = new Message(
      getName(MessageEnum.NavigationPayLoadMessage)
    );
    raiseMessage.addData(
      getName(MessageEnum.AuthTokenDataMessage),
      from.getData(getName(MessageEnum.AuthTokenDataMessage))
    );

    to.addData(getName(MessageEnum.NavigationRaiseMessage), raiseMessage);

    return to;
  };

  receive(from: string, message: Message): void {
    this.send(this.convert(message));
  }
}
