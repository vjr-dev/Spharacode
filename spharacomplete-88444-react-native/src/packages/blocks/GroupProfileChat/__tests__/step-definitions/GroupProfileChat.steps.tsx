//@ts-nocheck
//@ts-ignore
import { defineFeature, loadFeature } from "jest-cucumber";
import { shallow, ShallowWrapper } from "enzyme";
import * as helpers from "../../../../framework/src/Helpers";
import React from "react";
import GroupProfileChat from "../../src/GroupProfileChat";
import { Platform, View } from "react-native";
import { Message } from "../../../../framework/src/Message";
import MessageEnum, {
  getName,
} from "../../../../framework/src/Messages/MessageEnum";
import { runEngine } from "../../../../framework/src/RunEngine";

const screenProps = {
  navigation: {
    navigate: jest.fn(),
  },
  instance: { renderItem: jest.fn() },
  id: "GroupProfileChat",
};

const feature = loadFeature(
  "./__tests__/features/GroupProfileChat-scenario.feature"
);

defineFeature(feature, (test) => {
  beforeEach(() => {
    jest.resetModules();
    jest.doMock("react-native", () => ({ Platform: { OS: "web" } }));
    jest.spyOn(helpers, "getOS").mockImplementation(() => "web");
  });

  test("User navigates to GroupProfileChat", ({ given, when, then }) => {
    let exampleBlockA: ShallowWrapper;
    let instance: GroupProfileChat;

    given("I am a User loading GroupProfileChat", () => {
      exampleBlockA = shallow(<GroupProfileChat {...screenProps} />);
    });

    when("I navigate to the GroupProfileChat", () => {
      instance = exampleBlockA.instance() as GroupProfileChat;
    });

    then("GroupProfileChat will load with out errors", () => {
      expect(exampleBlockA).toBeTruthy();
      expect(exampleBlockA).toMatchSnapshot();
      const userListApi: Message = new Message(
        getName(MessageEnum.AccoutLoginSuccess)
      );
      instance.apiUserList = userListApi.messageId;
      runEngine.sendMessage("Unit Test", userListApi);
    });

    then("I can select the back button with out errors", () => {
      let buttonComponent = exampleBlockA.findWhere(
        (node) => node.prop("testID") === "backBtn"
      );
      buttonComponent.simulate("press");
      expect(buttonComponent).toBeTruthy();
    });

    then("I can select the option button with out errors", () => {
      let buttonComponent = exampleBlockA.findWhere(
        (node) => node.prop("testID") === "optionBtn"
      );
      instance.setState({ modal2Visible: true });
      buttonComponent.simulate("press");
      expect(buttonComponent).toBeTruthy();
    });

    then("I can select the cancle button with out errors", () => {
      let buttonComponent = exampleBlockA.findWhere(
        (node) => node.prop("testID") === "cancleModalBtn"
      );
      instance.setState({ modal2Visible: false });
      buttonComponent.simulate("press");
      expect(buttonComponent).toBeTruthy();
    });

    then("I can toggle the switch with out errors", () => {
      let switchComponent1 = exampleBlockA.findWhere(
        (node) => node.prop("testID") === "switch1"
      );
      instance.setState({ switch1: true });
      switchComponent1.props().onValueChange(true);
      instance.setState({ switch1: false });
      switchComponent1.props().onValueChange(false);

      Platform.OS = "android";

      switchComponent1.simulate("press");
      expect(switchComponent1).toBeTruthy();

      let switchComponent2 = exampleBlockA.findWhere(
        (node) => node.prop("testID") === "switch2"
      );
      instance.setState({ switch2: true });
      switchComponent2.props().onValueChange(true);

      instance.setState({ switch2: false });
      switchComponent2.props().onValueChange(false);

      Platform.OS = "android";

      switchComponent2.simulate("press");
      expect(switchComponent2).toBeTruthy();
    });

    then("I can see the Group Admins with out errors", () => {
      let gropuAdmins = exampleBlockA.findWhere(
        (node) => node.prop("testID") === "gropuAdmins"
      );
      expect(gropuAdmins).toBeTruthy();

      const item = gropuAdmins.renderProp("renderItem")({
        item: {
          id: 1,
          img: "https://randomuser.me/api/portraits/men/84.jpg",
          name: "James",
        },
      });
      expect(item).toMatchSnapshot();
      let renderItemWrapper = shallow(
        <View>
          {instance.renderItem({
            id: 1,
            img: "https://randomuser.me/api/portraits/men/84.jpg",
            name: "James",
          })}
        </View>
      );
      expect(renderItemWrapper).toBeTruthy();
      let renderItemFooterWrapper = shallow(
        <View>{instance.ListFooterComponent()}</View>
      );
      let addNewGroupAdminsBtn = renderItemFooterWrapper.findWhere(
        (node) => node.prop("testID") === "addNewGroupAdminsBtn"
      );
      addNewGroupAdminsBtn.simulate("press");
      expect(addNewGroupAdminsBtn).toBeTruthy();
    });

    then("I can see the Group Participants with out errors", () => {
      let groupParticipants = exampleBlockA.findWhere(
        (node) => node.prop("testID") === "groupParticipants"
      );
      expect(groupParticipants).toBeTruthy();

      const item = groupParticipants.renderProp("renderItem")({
        item: {
          id: 1,
          img: "https://randomuser.me/api/portraits/men/84.jpg",
          name: "James",
        },
      });
      expect(item).toMatchSnapshot();
      let renderItemWrapper = shallow(
        <View>
          {instance.renderItem({
            id: 1,
            img: "https://randomuser.me/api/portraits/men/84.jpg",
            name: "James",
          })}
        </View>
      );
      expect(renderItemWrapper).toBeTruthy();
      let renderItemFooterWrapper = shallow(
        <View>{instance.ListFooterComponent()}</View>
      );
      let addNewGroupAdminsBtn = renderItemFooterWrapper.findWhere(
        (node) => node.prop("testID") === "addNewGroupAdminsBtn"
      );
      addNewGroupAdminsBtn.simulate("press");
      expect(addNewGroupAdminsBtn).toBeTruthy();
    });

    then("I can leave the screen with out errors", () => {
      instance.componentWillUnmount();
      expect(exampleBlockA).toBeTruthy();
    });
  });
});
