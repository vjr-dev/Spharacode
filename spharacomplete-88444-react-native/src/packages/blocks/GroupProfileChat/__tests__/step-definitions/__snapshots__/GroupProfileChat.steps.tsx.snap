// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupProfileChat User navigates to GroupProfileChat 1`] = `
<ImageBackground
  source={
    Object {
      "testUri": "../../../packages/blocks/GroupProfileChat/assets/image_background-1.png",
    }
  }
  style={
    Object {
      "flex": 1,
      "height": 1334,
      "width": 750,
    }
  }
>
  <ImageBackground
    source={
      Object {
        "testUri": "../../../packages/blocks/GroupProfileChat/assets/image_background-2.png",
      }
    }
    style={
      Object {
        "flex": 1,
        "height": 1334,
        "width": 750,
      }
    }
  >
    <SafeAreaView
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <ScrollViewMock
        showsVerticalScrollIndicator={false}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "width": 750,
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "padding": 20,
              }
            }
            testID="backBtn"
          >
            <Image
              source={
                Object {
                  "testUri": "../../../packages/blocks/GroupProfileChat/assets/image_back.png",
                }
              }
              style={
                Object {
                  "height": 20,
                  "tintColor": "rgb(255,255,255)",
                  "width": 20,
                }
              }
            />
          </TouchableOpacity>
          <View
            style={
              Object {
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "rgb(255,255,255)",
                  "fontSize": 15,
                  "fontWeight": "700",
                }
              }
            >
              Group Name
            </Text>
          </View>
          <TouchableOpacity
            activeOpacity={0.2}
            style={
              Object {
                "width": "15%",
              }
            }
          />
        </View>
        <View>
          <Image
            source={
              Object {
                "testUri": "../../../packages/blocks/GroupProfileChat/assets/profile.png",
              }
            }
            style={
              Object {
                "alignSelf": "center",
                "borderColor": "white",
                "borderRadius": 50,
                "borderWidth": 1,
                "height": 118,
                "marginVertical": 20,
                "width": 118,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignSelf": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": 750,
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#404040",
                "borderRadius": 19,
                "borderWidth": 1,
                "height": 38,
                "justifyContent": "center",
                "left": 100,
                "width": 38,
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../packages/blocks/GroupProfileChat/assets/video.png",
                }
              }
              style={
                Object {
                  "height": "60%",
                  "tintColor": "rgba(255,211,154,1)",
                  "width": "60%",
                }
              }
            />
          </TouchableOpacity>
          <TouchableOpacity
            activeOpacity={0.2}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#404040",
                "borderRadius": 19,
                "borderWidth": 1,
                "height": 38,
                "justifyContent": "center",
                "width": 38,
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../packages/blocks/GroupProfileChat/assets/call.png",
                }
              }
              style={
                Object {
                  "height": "60%",
                  "tintColor": "rgba(255,211,154,1)",
                  "width": "60%",
                }
              }
            />
          </TouchableOpacity>
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#404040",
                "borderRadius": 19,
                "borderWidth": 1,
                "height": 38,
                "justifyContent": "center",
                "right": 100,
                "width": 38,
              }
            }
            testID="optionBtn"
          >
            <Image
              source={
                Object {
                  "testUri": "../../../packages/blocks/GroupProfileChat/assets/option.png",
                }
              }
              style={
                Object {
                  "height": "60%",
                  "tintColor": "rgba(255,211,154,1)",
                  "width": "60%",
                }
              }
            />
          </TouchableOpacity>
        </View>
        <Text
          style={
            Object {
              "color": "#a1a1a1",
              "paddingBottom": 4,
              "paddingStart": 15,
              "paddingTop": 40,
            }
          }
        >
          Invite to group via link
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "#363636",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "rgba(255,211,154,1)",
                "paddingStart": 15,
              }
            }
          >
            http://sphara.app/Devops-Group
          </Text>
          <TouchableOpacity
            activeOpacity={0.2}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#404040",
                "borderRadius": 19,
                "borderWidth": 1,
                "height": 38,
                "justifyContent": "center",
                "marginVertical": 10,
                "right": 15,
                "width": 38,
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../packages/blocks/GroupProfileChat/assets/share.png",
                }
              }
              style={
                Object {
                  "height": "60%",
                  "tintColor": "rgba(255,211,154,1)",
                  "width": "60%",
                }
              }
            />
          </TouchableOpacity>
        </View>
        <Text
          style={
            Object {
              "color": "#a1a1a1",
              "paddingBottom": 4,
              "paddingStart": 15,
              "paddingTop": 20,
            }
          }
        >
          About Group
        </Text>
        <Text
          style={
            Object {
              "backgroundColor": "#363636",
              "color": "#a1a1a1",
              "paddingBottom": 3,
              "paddingStart": 15,
              "paddingTop": 20,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#a1a1a1",
              "paddingBottom": 4,
              "paddingStart": 15,
              "paddingTop": 27,
            }
          }
        >
          Settings
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "#363636",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginVertical": 15,
                "width": 750,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#cdcdcd",
                  "fontSize": 17,
                  "left": 15,
                }
              }
            >
              Mute Group
            </Text>
            <Switch
              ios_backgroundColor="#000"
              onValueChange={[Function]}
              style={
                Object {
                  "right": 10,
                  "transform": Array [
                    Object {
                      "scaleX": 0.8,
                    },
                    Object {
                      "scaleY": 0.8,
                    },
                  ],
                }
              }
              testID="switch1"
              thumbColor="rgb(255,255,255)"
              trackColor={
                Object {
                  "false": "rgb(0,0,0)",
                  "true": "rgba(255,138,73,1)",
                }
              }
              value={false}
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginVertical": 5,
                "width": 750,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#cdcdcd",
                  "fontSize": 17,
                  "left": 15,
                }
              }
            >
              Aggregate Notification
            </Text>
            <Switch
              ios_backgroundColor="#000"
              onValueChange={[Function]}
              style={
                Object {
                  "right": 10,
                  "transform": Array [
                    Object {
                      "scaleX": 0.8,
                    },
                    Object {
                      "scaleY": 0.8,
                    },
                  ],
                }
              }
              testID="switch2"
              thumbColor="rgb(255,255,255)"
              trackColor={
                Object {
                  "false": "rgb(0,0,0)",
                  "true": "rgba(255,138,73,1)",
                }
              }
              value={false}
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginVertical": 16,
                "width": 750,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#cdcdcd",
                  "fontSize": 17,
                  "left": 15,
                }
              }
            >
              Set Background
            </Text>
            <TouchableOpacity
              activeOpacity={0.2}
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "right": 17,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#cdcdcd",
                    "fontSize": 17,
                  }
                }
              >
                Default
              </Text>
              <Image
                source={
                  Object {
                    "testUri": "../../../packages/blocks/GroupProfileChat/assets/next.png",
                  }
                }
                style={
                  Object {
                    "alignSelf": "center",
                    "height": 22,
                    "marginLeft": 10,
                    "tintColor": "#cdcdcd",
                    "width": 22,
                  }
                }
              />
            </TouchableOpacity>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginVertical": 16,
                "width": 750,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#cdcdcd",
                  "fontSize": 17,
                  "left": 15,
                }
              }
            >
              Media
            </Text>
            <TouchableOpacity
              activeOpacity={0.2}
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "right": 17,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#cdcdcd",
                    "fontSize": 17,
                  }
                }
              >
                10 items 
              </Text>
              <Image
                source={
                  Object {
                    "testUri": "../../../packages/blocks/GroupProfileChat/assets/next.png",
                  }
                }
                style={
                  Object {
                    "alignSelf": "center",
                    "height": 22,
                    "marginLeft": 5,
                    "tintColor": "#cdcdcd",
                    "width": 22,
                  }
                }
              />
            </TouchableOpacity>
          </View>
        </View>
        <Text
          style={
            Object {
              "color": "#a1a1a1",
              "paddingBottom": 4,
              "paddingStart": 15,
              "paddingTop": 27,
            }
          }
        >
          Group Admins
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "#363636",
              "width": "100%",
            }
          }
        >
          <FlatList
            ListFooterComponent={
              <View
                style={
                  Object {
                    "justifyContent": "center",
                    "marginHorizontal": 20,
                  }
                }
              >
                <TouchableOpacity
                  activeOpacity={0.2}
                  onPress={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "borderColor": "rgb(163,164,165)",
                      "borderRadius": 50,
                      "borderWidth": 1,
                      "height": 107,
                      "justifyContent": "center",
                      "width": 107,
                    }
                  }
                  testID="addNewGroupAdminsBtn"
                >
                  <Text
                    style={
                      Object {
                        "color": "rgb(163,164,165)",
                        "fontSize": 39,
                      }
                    }
                  >
                    +
                  </Text>
                </TouchableOpacity>
                <Text
                  style={
                    Object {
                      "alignSelf": "flex-end",
                      "color": "rgb(91,91,92)",
                      "paddingBottom": 10,
                      "paddingTop": 6,
                    }
                  }
                >
                  Add new
                </Text>
              </View>
            }
            data={
              Array [
                Object {
                  "id": 1,
                  "img": "https://randomuser.me/api/portraits/men/84.jpg",
                  "name": "James",
                },
                Object {
                  "id": 2,
                  "img": "https://randomuser.me/api/portraits/men/43.jpg",
                  "name": "Prince",
                },
                Object {
                  "id": 3,
                  "img": "https://randomuser.me/api/portraits/men/35.jpg",
                  "name": "Henry",
                },
                Object {
                  "id": 4,
                  "img": "https://randomuser.me/api/portraits/men/19.jpg",
                  "name": "Charls",
                },
                Object {
                  "id": 5,
                  "img": "https://randomuser.me/api/portraits/men/92.jpg",
                  "name": "David",
                },
              ]
            }
            disableVirtualization={false}
            horizontal={true}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onEndReachedThreshold={2}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEnabled={true}
            scrollEventThrottle={50}
            showsHorizontalScrollIndicator={false}
            style={
              Object {
                "marginTop": 20,
              }
            }
            testID="gropuAdmins"
            updateCellsBatchingPeriod={50}
            windowSize={21}
          />
        </View>
        <Text
          style={
            Object {
              "color": "#a1a1a1",
              "paddingBottom": 4,
              "paddingStart": 15,
              "paddingTop": 27,
            }
          }
        >
          Group Participants
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "#363636",
              "width": "100%",
            }
          }
        >
          <FlatList
            ListFooterComponent={
              <View
                style={
                  Object {
                    "justifyContent": "center",
                    "marginHorizontal": 20,
                  }
                }
              >
                <TouchableOpacity
                  activeOpacity={0.2}
                  onPress={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "borderColor": "rgb(163,164,165)",
                      "borderRadius": 50,
                      "borderWidth": 1,
                      "height": 107,
                      "justifyContent": "center",
                      "width": 107,
                    }
                  }
                  testID="addNewGroupAdminsBtn"
                >
                  <Text
                    style={
                      Object {
                        "color": "rgb(163,164,165)",
                        "fontSize": 39,
                      }
                    }
                  >
                    +
                  </Text>
                </TouchableOpacity>
                <Text
                  style={
                    Object {
                      "alignSelf": "flex-end",
                      "color": "rgb(91,91,92)",
                      "paddingBottom": 10,
                      "paddingTop": 6,
                    }
                  }
                >
                  Add new
                </Text>
              </View>
            }
            data={
              Array [
                Object {
                  "id": 1,
                  "img": "https://randomuser.me/api/portraits/men/84.jpg",
                  "name": "James",
                },
                Object {
                  "id": 2,
                  "img": "https://randomuser.me/api/portraits/men/43.jpg",
                  "name": "Prince",
                },
                Object {
                  "id": 3,
                  "img": "https://randomuser.me/api/portraits/men/35.jpg",
                  "name": "Henry",
                },
                Object {
                  "id": 4,
                  "img": "https://randomuser.me/api/portraits/men/19.jpg",
                  "name": "Charls",
                },
                Object {
                  "id": 5,
                  "img": "https://randomuser.me/api/portraits/men/92.jpg",
                  "name": "David",
                },
              ]
            }
            disableVirtualization={false}
            horizontal={true}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onEndReachedThreshold={2}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEnabled={true}
            scrollEventThrottle={50}
            showsHorizontalScrollIndicator={false}
            style={
              Object {
                "marginTop": 20,
              }
            }
            testID="groupParticipants"
            updateCellsBatchingPeriod={50}
            windowSize={21}
          />
        </View>
        <TouchableOpacity
          activeOpacity={0.2}
          style={
            Object {
              "backgroundColor": "#363636",
              "height": 98,
              "justifyContent": "center",
              "marginVertical": 59,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgb(255,157,43)",
                "fontSize": 17,
                "fontWeight": "700",
                "paddingLeft": 15,
              }
            }
          >
            Report
          </Text>
        </TouchableOpacity>
      </ScrollViewMock>
    </SafeAreaView>
  </ImageBackground>
</ImageBackground>
`;

exports[`GroupProfileChat User navigates to GroupProfileChat 2`] = `
<View
  style={
    Object {
      "marginLeft": 20,
    }
  }
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "borderColor": "rgb(255,255,255)",
        "borderRadius": 50,
        "borderWidth": 1,
        "height": 107,
        "width": 107,
      }
    }
  />
  <Text
    numberOfLines={1}
    style={
      Object {
        "alignSelf": "center",
        "color": "rgb(91,91,92)",
        "marginTop": 10,
        "textAlign": "center",
        "width": 118,
      }
    }
  />
</View>
`;

exports[`GroupProfileChat User navigates to GroupProfileChat 3`] = `
<View
  style={
    Object {
      "marginLeft": 20,
    }
  }
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "borderColor": "rgb(255,255,255)",
        "borderRadius": 50,
        "borderWidth": 1,
        "height": 107,
        "width": 107,
      }
    }
  />
  <Text
    numberOfLines={1}
    style={
      Object {
        "alignSelf": "center",
        "color": "rgb(91,91,92)",
        "marginTop": 10,
        "textAlign": "center",
        "width": 118,
      }
    }
  />
</View>
`;
